tools/clang/tools/dxc/dxclib/dxc.cpp

```diff
@@ -1011,6 +1011,27 @@ bool GetDLLFileVersionInfo(const char *dllPath, unsigned int *version) {
   return false;
 }
 
+bool GetDLLProductVersionInfo(const char *dllPath, std::string &productVersion) {
+  // This function is used to get product version information from the DLL file.
+  // This information in is not available through a Unix interface.
+#ifdef _WIN32
+  DWORD dwVerHnd = 0;
+  DWORD size = GetFileVersionInfoSize(dllPath, &dwVerHnd);
+  if (size == 0) return false;
+  std::unique_ptr<int[]> VfInfo(new int[size/sizeof(int)]);
```

Comment:
This fix is not correct as it will round down the number of bytes to the nearest multiple of four, but then you still pass  "size" as the size. You can make this a char or BYTE, all of the functions you pass this as an argument to take a void* anyways so they don't care. What was the reason for making it an int?

Reply:
Right, sorry ;) There is no reason for this to be int. I have copied that from another place in this file. I'll fix both.